api/v1alpha1/effectiveschedule_types.go

```diff
@@ -0,0 +1,59 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type (
+	// +kubebuilder:object:root=true
+	// +kubebuilder:subresource:status
+	// +kubebuilder:printcolumn:name="Schedule Namespace",type="string",JSONPath=`.spec.effectiveSchedules[0].namespace`,description="Schedule Namespace"
+	// +kubebuilder:printcolumn:name="Schedule Name",type="string",JSONPath=`.spec.effectiveSchedules[0].name`,description="Schedule Name"
+	// +kubebuilder:printcolumn:name="Age",type="date",JSONPath=".metadata.creationTimestamp"
+
+	// EffectiveSchedule is the Schema to persist schedules generated from Randomized schedules.
+	EffectiveSchedule struct {
+		metav1.TypeMeta   `json:",inline"`
+		metav1.ObjectMeta `json:"metadata,omitempty"`
+
+		Spec   EffectiveScheduleSpec   `json:"spec,omitempty"`
+		Status EffectiveScheduleStatus `json:"status,omitempty"`
+	}
+
+	// +kubebuilder:object:root=true
+
+	// EffectiveScheduleList contains a list of EffectiveSchedule
+	EffectiveScheduleList struct {
+		metav1.TypeMeta `json:",inline"`
+		metav1.ListMeta `json:"metadata,omitempty"`
+		Items           []EffectiveSchedule `json:"items"`
+	}
+
+	// EffectiveScheduleSpec defines the desired state of EffectiveSchedule
+	EffectiveScheduleSpec struct {
+
+		// EffectiveSchedules holds a list of effective schedules. The list may omit entries that aren't generated from
+		// smart schedules.
+		EffectiveSchedules []JobRef `json:"effectiveSchedules,omitempty"`
+	}
+
+	// JobRef represents a reference to a job in a Schedule object
+	JobRef struct {
+		Name      string             `json:"name,omitempty"`
+		Namespace string             `json:"namespace,omitempty"`
+		JobType   JobType            `json:"jobType,omitempty"`
+		Schedule  ScheduleDefinition `json:"schedule,omitempty"`
```

Comment:
Why do we need the schedule definition, if we have the job ref?

Reply:
This is the generated, effective Schedule for that job type, e.g. `@hourly-random` translated to `23 * * * *`.
But maybe `EffectiveSchedule` is a better name?

Reply:
Makes sense. Maybe GeneratedSchedule to not confuse it with the EffectiveSchedule definition?